src/sys/mman.rs

```diff
@@ -324,6 +324,21 @@ pub fn munlockall() -> Result<()> {
     unsafe { Errno::result(libc::munlockall()) }.map(drop)
 }
 
+#[cfg(any(target_os = "linux"))]
+pub type mincore_vec_char_t = std::os::raw::c_uchar;
+#[cfg(any(target_os = "freebsd", target_os = "macos"))]
+pub type mincore_vec_char_t = std::os::raw::c_char;
+
+/// Inspect page residency in memory at the address `addr` continuing for `length` bytes.
+///
+/// # Safety
+///
+/// `addr` and `vec` must meet all the requirements described in the `mincore(2)` man page.
+#[cfg(any(target_os = "linux", target_os = "freebsd", target_os = "macos"))]
+pub fn mincore(addr: *mut c_void, length: size_t, vec: *mut mincore_vec_char_t) -> Result<()> {
```

Comment:
Returning via a mutable pointer is not very Rusty.  It's also `unsafe`, because mincore has no way to know the lifetime of the pointed-to data.  Instead, you should return the data as a Vec.  You should also define an enum for the contents, instead of using c_char.  Finally, there's no reason why `addr` needs to be mutable.  So the signature should look like this:
```rust
libc_bitflags! {
    pub struct MincoreFlags {
        MINCORE_INCORE,
        ...
    }
}
pub fn mincore(addr: *const c_void, length: usize) -> Result<Vec<MincoreFlags>>
```

Reply:
> So the signature should look like this

I see, the new signature looks much safe and easy to use.
Let me try.